Create 2018-10-17-lesson-dev-plans.md (#262)
carpentries/carpentries.org: _posts/2018/10/2018-10-17-lesson-dev-plans.md

---
layout: page
authors: ["Tracy Teal", "Francois Michonnaeu", "Erin Becker"]
teaser: "What we're planning to accomplish for our lessons in the rest of the year."
title: "Curriculum Development Goals For the Rest of 2018"
date: 2018-10-17
time: "09:00:00"
tags: ["Lessons", "Curriculum"]
---

As we are entering the last quarter of 2018, and following up on our 
[Curriculum Development Vision post](https://carpentries.org/blog/2018/07/curriculum-vision/), 

> we want to outline our plans for the rest of the year and the beginning of 2019. 

Reviewer: ```suggestion
we want to outline plans for the rest of the year and the beginning of 2019. 
```

We plan to post regular updates on our curriculum development plans to allow community members to participate, 
get involved, and prepare planning for lessons or activities they want to engage in. If you are interested 
in participating in any of these projects, please send an email to [team@carpentries.org](mailto:team@carpentries.org).

## Plans for the 4th Quarter of 2018

### The R Genomics lesson

Data Carpentry initially published a curriculum for Genomics in 
[November 2017](https://datacarpentry.org/blog/2017/11/genomics-lesson-release). 
That workshop covers tabular metadata organization in spreadsheets, an introduction to the Unix bash shell, 

> FASTQ file processing with bash scripting, and connecting to AWS cloud instances. We are now preparing to publish 

Reviewer: ```suggestion
FASTQ file processing with bash scripting, and connecting to AWS cloud instances. The Genomics Curriculum Advisory Committee and a task force of others including ... are now preparing to publish 
```

a new version of this curriculum that includes data analysis and visualization in R. For this new version, the shell 
lessons will be simplified to free up time for the R lesson. This new release will correspond with an update to the existing 

> lessons to transition to longer-read data and a more modern aligner. We anticipate these lessons to be ready for teaching in early 2019. 

Reviewer: ```suggestion
lessons to transition to longer-read data and a more modern aligner. These lessons should be ready for teaching in early 2019. 
```

### Clarifying process for recurring releases of existing lessons

We have historically re-released all published Software Carpentry and Data Carpentry lessons periodically to give credit to new 
contributors and provide a stable DOI for the updated lessons. As our lesson stack has grown, it has become more time-consuming 
to publish lessons using our existing process, making it prohibitive to release lessons in a timely manner. During the last quarter 

> of 2018, we will be revamping our process for periodic lesson releases to ensure that we are able to continue to publish all of our 

Reviewer: ```suggestion
of 2018, a group from the Lesson Infrastructure committee will be revamping our process for periodic lesson releases to ensure that we are able to continue to publish all of our 
```

lessons at regular intervals. With this new process, we hope to restart regular lesson publications in the first quarter of 2019. 

### Lesson onboardings

> With the first series of Geospatial onboarding completed, we are starting to plan onboarding for Instructors interested in 

Reviewer: A bit of info on what onboardings are.

The first series of Geospatial onboardings has been complete. Over .. people attended these sessions! Video recordings of these sessions are available, so if you missed the onboardings, but are interested in teaching these lessons you can prepare by watching them.  Already (number of) Geospatial pilot workshops are being scheduled. We'll continue to get feedback on the lessons through the pilots. 

Onboardings for the Social Science curriculum are now beginning, and you can sign up for the onboarding on this etherpad. A recorded onboarding for this curriculum will be available after 9 November. 

Once the updated Genomics lessons are released, we will also start onboarding sessions 
for these lessons.

The Data Carpentry Geospatial, Social Science and Genomics Curriculum Advisory Committees, contributors and maintainers have done an amazing job developing these lessons from conceptualization to documentation and exercises. Overall there have been more than .. contributors to these lessons, and this important work will continue as onboardings and pilots continue.

teaching the R-based [Social Science curriculum](https://datacarpentry.org/blog/2018/05/social-sciences-publication) that 
was released in May. You can sign up for onboarding on [this Etherpad](https://pad.carpentries.org/socialsciences-onboarding). 
A recorded onboarding will be available after 9 November. A [video onboarding](https://www.youtube.com/watch?v=Qtnb_eeHt7E) is 
already available for the Geospatial lessons. Once the updated Genomics lessons are released, we will also start onboarding sessions 
for these lessons.

### Lesson template

> We will be continuing to work will continue on updating the lesson template. This includes integrating feedback on useful 

Reviewer: The Lesson Infrastructure committee will continue to work...

features such as [exercises that can be separated from lessons](https://carpentries.org/blog/2018/09/teaching-tip-exercise-discussion/) 
and [other proposed improvements](https://github.com/carpentries/styles/issues). We will also be undertaking a large-scale 
re-envisioning of the lesson template, to make it more flexible and easy-to-use. This project has been funding by the 
[R consortium](https://www.r-consortium.org/) and is described 
[in this proposal](https://github.com/carpentries/2018-r-consortium-lessondown-proposal/blob/master/proposal.Rmd). 

## Tentative Plans for the 1st Quarter of 2019

### Data Carpentry Social Sciences Python and SQL

There are “alpha” versions of the lessons for 
[Python](http://www.datacarpentry.org/python-socialsci/) and 
[SQL](http://www.datacarpentry.org/sql-socialsci/) for the Data Carpentry Social Science curriculum. 
We will work with the community to make these lessons ready for an initial publication in early 2019.

### Library Carpentry

> On the path of Library Carpentry being integrated as an official Lesson Program within The Carpentries, 

Reviewer: ```suggestion
On the path of [Library Carpentry incubation](https://carpentries.org/blog/2018/10/library-carpentry-next-steps/) as a Lesson Program within The Carpentries, 
```

> we will work on releasing officially an initial set of lessons. The lessons that will be included in this initial 

Reviewer: ```suggestion
we will work with the Library Carpentry Curriculum Advisory Committee and maintainers on releasing officially an initial set of lessons. The lessons that will be included in this initial 
```
